src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -85,36 +80,89 @@ DatabaseShardImp::init()
             "Already initialized";
         return false;
     }
-    if (ledgersPerShard_ == 0 || ledgersPerShard_ % 256 != 0)
+
+    auto fail = [&](std::string const& msg)
     {
         JLOG(j_.error()) <<
-            "ledgers_per_shard must be a multiple of 256";
+            "[" << ConfigSection::shardDatabase() << "] " << msg;
         return false;
+    };
+
+    Config const& config {app_.config()};
+    Section const& section {config.section(ConfigSection::shardDatabase())};
+    if (section.empty())
+        return fail("missing configuration");
+
+    {
+        // Node and shard stores must use same earliest ledger sequence
+        std::uint32_t seq;
+        if (get_if_exists<std::uint32_t>(
+                config.section(ConfigSection::nodeDatabase()),
+                "earliest_seq",
+                seq))
+        {
+            std::uint32_t seq2;
+            if (get_if_exists<std::uint32_t>(section, "earliest_seq", seq2) &&
+                seq != seq2)
+            {
+                return fail("and [" + ConfigSection::shardDatabase() +
+                    "] both define 'earliest_seq'");
+            }
+        }
+    }
+
+    if (!get_if_exists<boost::filesystem::path>(section, "path", dir_))
+        return fail("'path' missing");
+
+    if (boost::filesystem::exists(dir_))
+    {
+        if (!boost::filesystem::is_directory(dir_))
+            return fail("'path' must be a directory");
+    }
+    else
+        boost::filesystem::create_directories(dir_);
+
+    if (!get_if_exists<std::uint64_t>(section, "max_size_gb", maxDiskSpace_))
+        return fail("'max_size_gb' missing");
+
+    if (maxDiskSpace_ < minDiskSpace_)
```

Comment:
is `minDiskSpace` also in GB? does it need to be converted to bytes too?

Reply:
Yes! thanks. :+1:

Reply:
Might be helpful to use labels for units -> `minDiskSpace_` means it's in bytes, `minDiskSpaceGB_` means it's in gigabytes... since we don't have any types for these units afaik. Maybe we need some std::chrono-like system :)

Reply:
Bah, actually `minDiskSpace ` _is_ always in GB and before the conversion (at this point), so is `maxDiskSpace_`. So the code is correct as is. I am not a fan of the label idea but using a temp here and assigning a converted value to `maxDiskSpace_` will help avoid confusion.  I'll also narrow `minDiskSpace`'s scope to be local since its only used in one place.